osrf/vrx
# Fix timeout issue

I introduced an issue in #449 changing the default value of `timeoutScore` from `-1` to `200`. See how we're using `timeoutScore` in [scoring_plugin.cc::OnFinished()](https://github.com/osrf/vrx/blob/master/vrx_gazebo/src/scoring_plugin.cc#L278-L281). 

The original intent of that block of code is to update the current score to the `timeoutScore` only when `timeoutScore` is set. We were assuming that if `timeoutScore` isn't set, its value will be negative. Our change in #449 breaks this convention causing the score to always be updated to `timeoutScore` when a task timeouts. This is problematic in tasks such as stationkeeping or wayfinding because the task will always timeout.

How to test it?

1. Launch a stationkeeping task (**update the world path**) and verify that after the timeout, the score is reasonable (not `200`):

```
roslaunch vrx_gazebo vrx.launch world:=/home/caguero/vrx2022_phase3/src/vrx-docker/generated/task_generated/stationkeeping/worlds/stationkeeping0.world verbose:=true
```
```
rostopic echo /vrx/task/info
```

2.1 Launch a gymkhana task (**update the world path**):
```
roslaunch vrx_gazebo vrx.launch world:=/home/caguero/vrx2022_phase3/src/vrx-docker/generated/task_generated/stationkeeping/worlds/stationkeeping0.world verbose:=true
```
```
roslaunch vrx_gazebo usv_joydrive.launch
```
```
rostopic echo /vrx/task/info
```

Verify that crossing the channel and letting the task timeout you get a value lower than `200`.

2.2 Launch a gymkhana task as 2.1:

Verify that leaving the task to timeout without crossing the channel returns a score of `200`.

2.3 Launch a gymkhana task as 2.1:

Verify that making an illegal move (crossing a gate backwards or skipping a gate) terminates the run and returns a score of `200`.
